sympy/combinatorics/pc_groups.py

```diff
@@ -0,0 +1,494 @@
+
+        >>> F, x1, x2 = free_group("x1, x2")
+        >>> pc_relators = {x1**2 : 1, x1*x2*x1**-1 : x2**-1, x1**-1*x2*x1 : x2**-1}
+        >>> relative_order = [2, None]
+        >>> word = x2**2*x1**7
+        >>> group = word.group
+        >>> collector = Collector(pc_relators, relative_order, group)
+        >>> power_rel, conj_rel = collector.relations()
+        >>> power_rel
+        {x1**2: 1}
+        >>> conj_rel
+        {x1**-1*x2*x1: x2**-1, x1*x2*x1**-1: x2**-1}
+
+        """
+        power_relators = {}
+        conjugate_relators = {}
+        for key, value in self.pc_relators.items():
+            if len(key.array_form) == 1:
+                power_relators[key] = value
+            else:
+                conjugate_relators[key] = value
+        return power_relators, conjugate_relators
+
+    def subword_index(self, word, w):
+        """
+        Returns the start and ending index of a given
+        subword in a word.
+
+        Examples
+        ========
+        >>> from sympy.combinatorics.pc_groups import Collector
+        >>> from sympy.combinatorics.free_groups import free_group
+        >>> F, x1, x2 = free_group("x1, x2")
+        >>> pc_relators = {x1**2 : 1, x1*x2*x1**-1 : x2**-1, x1**-1*x2*x1 : x2**-1}
+        >>> relative_order = [2, None]
+        >>> word = x2**2*x1**7
+        >>> group = word.group
+        >>> collector = Collector(pc_relators, relative_order, group)
+        >>> w = x2**2*x1
+        >>> collector.subword_index(word, w)
+        (0, 3)
+        >>> w = x1**7
+        >>> collector.subword_index(word, w)
+        (2, 9)
+
+        """
+        low = -1
+        high = -1
+        for i in range(len(word)-len(w)+1):
+            if word.subword(i, i+len(w)) == w:
+                low = i
+                high = i+len(w)
+                break
+        if low == high == -1:
+            return -1, -1
+        return low, high
+
+    def map_relation(self, w):
+        """
+        Return a conjugate relation.
+        Given a word formed by two free group elements, the
+        corresponding conjugate relation with those free
+        group elements is formed and mapped with the collected
+        word in the polycyclic presentation.
+
+        Examples
+        ========
+        >>> from sympy.combinatorics.pc_groups import Collector
+        >>> from sympy.combinatorics.free_groups import free_group
+        >>> F, x0, x1, x2 = free_group("x0, x1, x2")
+        >>> pc_relators = {x0**-1*x1*x0: x1**2, x1**-1*x2*x1: x2, x0**-1*x2*x0: x2*x1}
+        >>> relative_order = [2, 2, 3]
+        >>> word = x2**2*x1**7
+        >>> group = word.group
+        >>> collector = Collector(pc_relators, relative_order, group)
+        >>> w = x2*x1
+        >>> collector.map_relation(w)
+        x2
+        >>> w = x1*x0
+        >>> collector.map_relation(w)
+        x1**2
+
+        """
+        group = w.group
+        gens = sorted(w.contains_generators())
```

Comment:
Can we trust that the sort order is always what we want? If `w.array_form` has two items, could we just take them in the that order?

Reply:
I think we can trust it, sort will maintain the order of free groups mentioned by the user. We can't take directly from `w.array_form` because conjugate relations should be in the form `x{j}^-1*x{i}*x{j}` where `j`must be less than `i`.

Reply:
It was my understanding that this would be called only for words `w = xi**ei*xj**ej` where we already know that `j < i`. (As `xi**ei*xj` is a minimal uncollected word. The order was determined by looking at the indices.) Is there any reason to call this for other kinds of words `w`?

Reply:
Ahh!!, that's right they are already sorted.